### ISSUE TITLE
Cannot deploy with hardhat reef

### ISSUE THREAD

**Issue by dillon-co** [reef-defi/hardhat-reef#15]
I'm getting this error when I try to deploy the flipper script in the `hardhat-reef-example` repo

Is this just a versioning thing?

```An unexpected error occurred:

/Users/dilloncortez/workspace/reef/hardhat-reef-examples/node_modules/@reef-defi/hardhat-reef/dist/src/index.js:16
    const userPath = userConfig.paths?.newPath;
                                      ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:1053:16)
    at Module._compile (internal/modules/cjs/loader.js:1101:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/dilloncortez/workspace/reef/hardhat-reef-examples/hardhat.config.js:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)

**Comment by Frenkiee** [comment]
Hi, thanks for reporting the problem! 
There was an error in the library, which is now fixed with version `"@reef-defi/hardhat-reef": "^0.4.18"`. 
Let me know if this resolved your problem.

**Comment by dillon-co** [comment]
I'm getting the same error on the same line (line 16) but it's saying `...userConfig.networks?.reef` is the culprit

**Comment by dillon-co** [comment]
I didn't change the `hardhat.config.js` and just left the default in when I pulled. Could that be the cause?

**Comment by dillon-co** [comment]
How is anyone supposed to build anything with reef if the example repo doesn't even work?

**Comment by jbargu** [comment]
**Doesn't work** is a rather bold statement. It does not work for **you**. For your convenience I cloned a fresh repo of `hardhat-reef-examples`, performed the following 4 steps:

1. Run `yarn`.
2. Run [a local reef-chain](https://github.com/reef-defi/reef-chain/commits/master) with `make run`.
3. Run `npx hardhat run scripts/erc20/deploy.js` - ERC20 was successfully deployed.
4. Run `npx hardhat run scripts/flipper/deploy.js`- Flipper contract was successfully deployed as well the contract state was flipped.

`yarn` and `node` versions:
```
/tmp/hardhat-reef-examples
$ yarn --version                                                                       master
2.4.1

/tmp/hardhat-reef-examples
$ node --version                                                                       master
v14.17.0
```
The OS was Ubuntu 20.04 LTS.

The error you get above seems like an issue with an incorrect compiler/environment, but then I would be guessing, as you did not provide any additional information, e.g. what OS are you using, your `node` version, your `yarn` version, which exact steps did you take to obtain the error - meaning exact commands that you have used: `when I try to deploy the flipper script` is not something I could deterministically replicate, would you agree? 

To sum up, the above is a **lazy question** and to help you with the issue, the following two references may guide on how to ask a good question:

1. [How To Ask Questions The Smart Way](http://catb.org/~esr/faqs/smart-questions.html)
2. [How do I ask a good question?](https://stackoverflow.com/help/how-to-ask)

It is a skill that will carry on and serve you well beyond this repo.

**Comment by matjazonline** [comment]
Maybe Node.js version is your issue. Optional chaining is currently not supported in Node.js version 13 and below.

**Comment by dillon-co** [comment]
My node version is 12.18.4. that's probably the issue. Thank you

**Comment by dillon-co** [comment]
@jbargu that was a pretty condescending way to say you need more info.

**Comment by dillon-co** [comment]
@jbargu you're right, i definitely could've provided more information. My mistake, I'm a blockchain noob.

Maybe just ask for more info next time instead of being a dick?
